lib/vm.py

```diff
@@ -287,6 +288,30 @@ def checkpoint(self) -> Snapshot:
                                                        'new-name-label': 'Checkpoint of %s' % self.uuid}),
                         self.host)
 
+    def connect_vdi(self, vdi: VDI, device: str = "autodetect") -> str:
+        logging.info(f">> Plugging VDI {vdi.uuid} on VM {self.uuid}")
+        vbd_uuid = self.host.xe("vbd-create", {
+            "vdi-uuid": vdi.uuid,
+            "vm-uuid": self.uuid,
+            "device": device,
+        })
+        self.host.xe("vbd-plug", {"uuid": vbd_uuid})
+
+        self.vdis.append(vdi)
+
+        return vbd_uuid
```

Comment:
if `vbd-plug` fails we may want to remove the vbd?

It looks like 2 fixtures for create and plug would be a good fit, except that it would not allow to store `self.vdis`, but I don't see why we need to store it (note: this advocates for finer commit granularity, where we have a better opportunity to explain why we're adding such a feature)... and creating a VDI object requires at least on XAPI call, which can make this sensibly slower especially for a VM with many VDIs